subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc

```diff
@@ -179,3 +179,11 @@ property instead.
 property has been deprecated.
 Use the link:{groovyDslPath}/org.gradle.api.tasks.scala.ScalaCompile.html#org.gradle.api.tasks.scala.ScalaCompile:destinationDirectory[ScalaCompile.destinationDirectory]
 property instead.
+
+[[upload_task_deprecation]]
+=== Deprecated `Upload` task
+
+Gradle used to have two way of publishing artifacts.
+Now, the situation has been cleared and all build should use the `maven-publish` plugin.
+The last remaining artifact of the old way of publishing is the `Upload` task that has been deprecated and scheduled for removal in Gradle 8.0.
+Existing clients should migrate to the `maven-publish` plugin.
```

Comment:
Maybe add a link to the `maven-publish` docs or sample from here?

Reply:
Done.